apis/filesystem/src/main/java/org/jclouds/filesystem/strategy/internal/FilesystemStorageStrategyImpl.java

```diff
@@ -154,32 +161,95 @@ public boolean createContainerInLocation(String container, Location location, Cr
    }
 
    @Override
-   public ContainerAccess getContainerAccess(String container) {
+   public synchronized ContainerAccess getContainerAccess(String container) {
       Path path = new File(buildPathStartingFromBaseDir(container)).toPath();
-      Set<PosixFilePermission> permissions;
-      try {
-         permissions = getPosixFilePermissions(path);
-      } catch (IOException ioe) {
-         throw Throwables.propagate(ioe);
+
+      if ( System.getProperty("os.name", "").toLowerCase().contains("windows") ) {
+         UserPrincipal everyone = null;
+         try {
+            everyone = getDefault().getUserPrincipalLookupService()
+                  .lookupPrincipalByName("Everyone");
+         } catch (IOException e) {
+            throw new RuntimeException(e);
+         }
+         AclFileAttributeView aclFileAttributes = java.nio.file.Files.getFileAttributeView(
+               path, AclFileAttributeView.class);
+         try {
+            for (AclEntry aclEntry : aclFileAttributes.getAcl()) {
+               if (aclEntry.principal().equals(everyone)) {
+                  return ContainerAccess.PUBLIC_READ;
+               }
+            }
+            return ContainerAccess.PRIVATE;
+         } catch (IOException e) {
+            throw new RuntimeException(e);
+         }
+      } else {
+         Set<PosixFilePermission> permissions;
+         try {
+            permissions = getPosixFilePermissions(path);
+         } catch (IOException ioe) {
+            throw Throwables.propagate(ioe);
+         }
+         return permissions.contains(PosixFilePermission.OTHERS_READ)
+               ? ContainerAccess.PUBLIC_READ : ContainerAccess.PRIVATE;
       }
-      return permissions.contains(PosixFilePermission.OTHERS_READ)
-            ? ContainerAccess.PUBLIC_READ : ContainerAccess.PRIVATE;
    }
 
    @Override
-   public void setContainerAccess(String container, ContainerAccess access) {
+   public synchronized void setContainerAccess(String container, ContainerAccess access) {
       Path path = new File(buildPathStartingFromBaseDir(container)).toPath();
-      Set<PosixFilePermission> permissions;
-      try {
-         permissions = getPosixFilePermissions(path);
-         if (access == ContainerAccess.PRIVATE) {
-            permissions.remove(PosixFilePermission.OTHERS_READ);
-         } else if (access == ContainerAccess.PUBLIC_READ) {
-            permissions.add(PosixFilePermission.OTHERS_READ);
+
+      if ( System.getProperty("os.name", "").toLowerCase().contains("windows") ) {
+         UserPrincipal everyone = null;
+         UserPrincipal me = null;
+         try {
+            everyone = getDefault().getUserPrincipalLookupService()
+                  .lookupPrincipalByName("Everyone");
+         } catch (IOException e) {
+            e.printStackTrace();
+            return;
```

Comment:
When reading, if this exception happens, we propagate an exception, but when writing, we just silently return and ignore it? Can we change one or the other to have a consistent behavior?